lib/grandpa/grandpa.go

```diff
@@ -1281,3 +1372,208 @@ func (s *Service) lenVotes(stage Subround) int {
 
 	return count
 }
+
+func (s *Service) handleVoteMessage(from peer.ID, vote *VoteMessage) {
+	logger.Debugf("received vote message, (peer: %s, msg: %+v)", from, vote)
+	s.sendTelemetryVoteMessage(vote)
+
+	v, err := s.validateVoteMessage(from, vote)
+	if err != nil {
+		logger.Debugf("failed to validate vote message {%v}: %s", vote, err)
+		return
```

Comment:
this should be an error returned and handled by the caller, not a debug log

Reply:
bump on this @EclesioMeloJunior